docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+
+```powershell
+tensorflow-ui-examples $ ayx_plugin_cli create-ayx-plugin
+Tool Name: TextClassifier
+Tool Type (input, multiple-inputs, multiple-outputs, optional, output, single-input-single-output, multi-connection-input-anchor) [optional]:
+Description []: Create, train, deploy, and use a Tensorflow based text classifier!
+Tool Version [1.0]:
+DCM Namespace []:
+Creating single-input-single-output plugin: Filter UI Tool
+[Create plugin] started
+[Create plugin] .  Create plugin
+[Create plugin] Installing UI components via npm
+[Create plugin] Creating Alteryx Plugin...
+// output truncated //
+[Generating test files for FilterUITool] finished
+```
+
+After the command finishes, you should have template files similar to below at the named locations for your UI and python SDK.
+
+#### 1.2.5.1. ui/TextClassifier/src/index.tsx
+```jsx
+const App = () => {
+  const classes = useStyles();
+  const [model, handleUpdateModel] = useContext(UiSdkContext);
+
+  useEffect(() => {
+    handleUpdateModel(model)
+  }, []);
+
+  return (
+    <Box p={4}>
+     <Grid container spacing={4} direction="column" alignItems="center">
+        <Grid item>
+          <Alteryx className={classes.alteryx} />
+        </Grid>
+        <Grid item>
+          <Typography variant="h3">
+            To get started, edit src/index.tsx
+          </Typography>
+        </Grid>
+      </Grid>
+    </Box>
+  )
+}
+
+const Tool = () => {
+  return (
+    <DesignerApi messages={{}}>
+      <AyxAppWrapper> 
+        <App />
+      </AyxAppWrapper>
+    </DesignerApi>
+  )
+}
+
+ReactDOM.render(
+  <Tool />,
+  document.getElementById('app')
+);
+```
+
+#### 1.2.5.2. backend/ayx_plugins/text_classifier.py
+```python
+class TextClassifier(PluginV2):
+    """A sample Plugin that passes data from an input connection to an output connection."""
+
+    def __init__(self, provider: AMPProviderV2):
+        """Construct the plugin."""
+        self.name = "TextClassifier"
+        self.provider = provider
+        self.provider.io.info(f"{self.name} tool started")
+
+    def on_record_batch(self, batch: "pa.Table", anchor: Anchor) -> None:
+        # truncated code
+
+    def on_incoming_connection_complete(self, anchor: Anchor) -> None:
+        # truncated code
+
+    def on_complete(self) -> None:
+        # truncated code
+```
+
+## 1.3. Backend: A Naive Tensorflow Text Classifier
+
+Here, we use the term naive to describe a tool that assumes perfect input in a known structure. 
+Later we will add input fields that accept arbitrary values. But we can start with controlled, known values first to simplify testing.
+To further simplify the transition to User populated or arbitrary data, we can also create a constants.py file.
+For now, leave it blank. We'll need it later to create static values for our tests.
```

Comment:
TW review:

Here, we use the term naive to describe a tool that assumes perfect input in a known structure. Later we will add input fields that accept arbitrary values. But we can start with controlled, known values first to simplify testing. To further simplify the transition to user-populated or arbitrary data, we can also create a constants.py file. For now, leave it blank. We'll need it later to create static values for our tests.